### DISCUSSION TITLE
Add support for mta-sts on external domains for users.

### DISCUSSION THREAD

**Discussion by c0nfigurati0n** [anonaddy/anonaddy#376] [Feature Requests]
@willbrowningme it might be interesting to implement the MTA Strict Transport Security protocol in the DNS setup area for users who use custom domains. So that way they don't have to pay for hosting and get the additional benefits of using the mta-sts protocol.

# additional resources
MTA-STS setup: https://www.hardenize.com/blog/mta-sts
MTA-STS Policy Delegation: https://datatracker.ietf.org/doc/html/rfc8461#section-8.2

**Comment by willbrowningme** [comment]
Thank you, this would be nice to add. I have looked into this in the past, however I cannot find a good solution for the `https://mta-sts.example.com/.well-known/mta-sts.txt` which must be served over https with a valid TLS certificate for example.com.

I don't really want to get involved with issuing and renewing thousands of letsencrypt certs for custom domains. It would mean either creating a new Nginx block or adding all domains to the same block too.

> Note that in all such cases, the policy endpoint
   ("https://mta-sts.user.example/.well-known/mta-sts.txt" in this
   example) must still present a certificate valid for the Policy Host
   ("mta-sts.user.example"), and not for that host at the provider's
   domain ("mta-sts.provider.example").

**Comment by c0nfigurati0n** [comment]
@willbrowningme yeah, i noticed this. I wonder if there is a way to bypass this?

**Reply by willbrowningme** [reply]
No way that I'm aware of unfortunately. I think renewing and issuing thousands of letsencrypt certs for custom domains would be a real headache and add a lot of maintenance overhead.

**Reply by c0nfigurati0n** [reply]
@willbrowningme is it not possible to automate this process? Maybe via a Script, API, etc, etc?
